boards/arm/96b_nitrogen/96b_nitrogen_defconfig

```diff
@@ -5,7 +5,7 @@ CONFIG_SOC_NRF52832_QFAA=y
 CONFIG_BOARD_96B_NITROGEN=y
 
 # Enable MPU
-CONFIG_ARM_MPU_NRF52X=y
+CONFIG_ARM_MPU=y
```

Comment:
why dont we just configure CONFIG_MPU and let the system figure out what implementation is being used, i.e. NXP vs ARM vs something else.

Reply:
This is exactly what this PR does, except the fact that it uses CONFIG_ARM_MPU instead of CPU_HAS_MPU as a master switch.

Unification with other architectures and introduction of CONFIG_MPU option might follow.